mongo/find.go

```diff
@@ -436,3 +441,44 @@ func encodeCursor(cursorData bson.D) (string, error) {
 	data, err := bson.Marshal(cursorData)
 	return base64.RawURLEncoding.EncodeToString(data), err
 }
+
+type (
+	Error struct {
+		message string
+	}
+)
+
+func (e *Error) Error() string {
+	return e.message
+}
+
+// validatePaginatedFields verifies that the results' struct has a bson tag that matches each paginated field
+func validatePaginatedFields(results interface{}, paginatedFields []string) error {
+	// results is a pointer to a slice (which is also a pointer) so we need to dereference twice
+	elem := reflect.TypeOf(results).Elem().Elem()
+	// We can't validate bson.Raw as we don't have the bson tags
+	if elem == reflect.TypeOf(bson.Raw{}) || elem == reflect.TypeOf(&bson.Raw{}) {
+		return nil
+	}
+	// the slice's type could be a pointer to a struct or a struct
+	if reflect.ValueOf(elem).Kind() == reflect.Ptr {
```

Comment:
elem.Kind == reflect.Ptr is enough here otherwise the condition will always be true.

testing with *[]model, elem.Kind is reflect.Struct., but reflect.ValueOf(elem).Kind() returns  reflect.Type which is an interface and underneath a ptr.

Reply:
Thank you, I'll make sure to add unit tests for this.